tests/integration/pusher/trigger.js

```diff
@@ -298,6 +300,76 @@ describe("Pusher", function() {
     });
   });
 
+  describe("Pusher with encryptionMasterKey", function() {
+    var pusher;
+
+    var testMasterKey = "01234567890123456789012345678901";
+  
+    beforeEach(function() {
+      pusher = new Pusher({ appId: 1234, key: "f00d", secret: "beef", encryptionMasterKey: testMasterKey });
+      nock.disableNetConnect();
+    });
+  
+    afterEach(function() {
+      nock.cleanAll();
+      nock.enableNetConnect();
+    });
+  
+    describe("#trigger", function() {
+      it("should not encrypt the body of an event triggered on a single channel", function(done) {
+        var mock = nock("http://api.pusherapp.com")
+          .filteringPath(function(path) {
+            return path
+              .replace(/auth_timestamp=[0-9]+/, "auth_timestamp=X")
+              .replace(/auth_signature=[0-9a-f]{64}/, "auth_signature=Y");
+          })
+          .post(
+            "/apps/1234/events?auth_key=f00d&auth_timestamp=X&auth_version=1.0&body_md5=e95168baf497b2e54b2c6cadd41a6a3f&auth_signature=Y",
+            { name: "my_event", data: "{\"some\":\"data \"}", channels: ["one"] }
+          )
+          .reply(200, "{}");
+
+        pusher.trigger("one", "my_event", { some: "data "}, done);
+      });
+      
+      it("should encrypt the body of an event triggered on a private-encrypted- channel", function(done) {
+        var sentPlaintext = "Hello!";
+
+        var mock = nock("http://api.pusherapp.com")
+          .filteringPath(function(path) {
+            return path
+              .replace(/auth_timestamp=[0-9]+/, "auth_timestamp=X")
+              .replace(/auth_signature=[0-9a-f]{64}/, "auth_signature=Y")
+              .replace(/body_md5=[0-9a-f]{32}/, "body_md5=Z");
+          })
+          .post(
+            "/apps/1234/events?auth_key=f00d&auth_timestamp=X&auth_version=1.0&body_md5=Z&auth_signature=Y",
+            function (body) {
+              if (body.name !== "test_event") return false;
+              if (body.channels.length !== 1) return false;
+              var channel = body.channels[0];
+              if (channel !== "private-encrypted-bla") return false;
+              var encrypted = JSON.parse(body.data);
+              var nonce = Buffer.from(encrypted.nonce, 'base64');
+              var ciphertext = Buffer.from(encrypted.ciphertext, 'base64');
+              var channelSharedSecret = pusher.channelSharedSecret(channel);
+              var receivedPlaintextBytes = nacl.secretbox.open(ciphertext, nonce, channelSharedSecret);
+              var decoder = new StringDecoder('utf8');
+              var receivedPlaintextJson = decoder.write(receivedPlaintextBytes);
+              console.log("1 received plaintext json", typeof(receivedPlaintextJson), JSON.stringify(receivedPlaintextJson));// debug travisci
```

Comment:
I'm guessing you are still working on this, so I'll hold off reviewing the tests.